Docs/docs/sdk/detecting-vrcsdk.md

```diff
@@ -0,0 +1,48 @@
+---
+title: "Detecting VRChat SDK"
+slug: "detecting-vrcsdk"
+hidden: false
+createdAt: "2023-10-31T07:18:15.860Z"
+updatedAt: "2023-10-31T07:18:15.860Z"
+---
+
+If your unity tools or libraries want to use APIs of VRChat SDK but not depends on VRChat SDK,
+there are several ways to detect VRChat SDK installation.
+
+## Using Version Defines (Recommended) {#using-version-defines}
+
+The best way to detect VRChat SDK installation is that using [Version Defines of assembly definition file][version-defines].
+
+You can define symbols in your assembly when `com.vrchat.base`, `com.vrchat.avatars`, or `com.vrchat.worlds` are installed.
+It's highly recommended to use Expression to define your symbol only when installed VRChat SDK version is compatible with your tool.
+For versioning of VRChat SDK, please refer [VCC Docs][versioning].
+
+Since Version Defines is feature for UPM packages, this method only works for VPM-based SDKs, which treated as UPM package by Unity.
+If you also want to detect legacy unitypackage-based SDKs, you can combine with the following legacy method by defining
+same symbol as VRCSDK defines or adding code like this onto evey file.
+
+```csharp
+#if !YOUR_VRCSDK3_AVATARS && !YOUR_VRCSDK3_WORLDS && VRC_SDK_VRCSDK3
+    #if UDON
+        #define YOUR_VRCSDK3_WORLDS
+    #else
+        #define YOUR_VRCSDK3_AVATARS
+    #endif
+#endif
+```
+
+[version-defines]: https://docs.unity3d.com/2019.4/Documentation/Manual/ScriptCompilationAssemblyDefinitionFiles.html#define-symbols
+[versioning]: https://vcc.docs.vrchat.com/vpm/packages/#brandingbreakingbumps
+
+## Using Legacy VRCSDK-defined scripting symbols (Deprecated) {#using-scripting-symbols}
+
+The other method to detect VRChat SDK installation is that using VRCSDK-defined scripting symbols.
+For all VRCSDK projects, `VRC_SDK_VRCSDK3` will be defined and for worlds projects, `UDON` will be defined.
+
+This method is legacy, obsoleted method that will be removed in the feature so please do not depend (only) on this method.
+
+In the old VRChat SDK, we use `VRC_SDK_VRCSDK3` and `UDON` symbol for internal usages but since those symbols are
+active whole project, many tools depends on those symbols for detecting VRChat SDK.
+Currently, in the VRChat SDK, all those symbol usages are migrated to VersionDefines.
+Please migrate to Version Defines as soon as possible!
+
```

Comment:
```suggestion
The other method to detect VRChat SDK installation is with VRCSDK-defined scripting symbols.
For all VRCSDK projects, `VRC_SDK_VRCSDK3` will be defined, and for world projects, `UDON` will be defined.

This method is deprecated and will be removed in the future. Do not solely depend on this method.

In the old VRChat SDK, `VRC_SDK_VRCSDK3` and `UDON` symbols were used internally. But because those symbols are active in the whole project, many tools depend on those symbols for detecting the VRChat SDK.

Currently, in the VRChat SDK, all those symbol usages are migrated to Version Defines. Please migrate to Version Defines as soon as possible!

```